gmso/external/convert_dataframe.py

```diff
@@ -0,0 +1,552 @@
+            ]
+            outList.extend(values_dictList)
+            columnsList.extend(keys)
+        elif isinstance(valuesList[0], u.unyt_array) and not isinstance(
+            valuesList[0], u.unyt_quantity
+        ):
+            outList.extend(np.array(valuesList).T)
+            if column == "position":
+                columnsList.extend(["x", "y", "z"])
+            else:
+                columnsList.extend(
+                    [f"{column}-({i})" for i in range(len(valuesList[0]))]
+                )
+        elif isinstance(valuesList[0], tuple) or isinstance(
+            valuesList[0], list
+        ):  # could be connection_members
+            outList.extend(np.array(valuesList).T)
+            if "connection_members" in column:
+                columnsList.extend(
+                    [
+                        f"{parameter} member ({i})"
+                        for i in range(len(valuesList[0]))
+                    ]
+                )
+            else:
+                columnsList.extend(
+                    [f"{column}-({i})" for i in range(len(valuesList[0]))]
+                )
+
+            # handle positions?
+            # handle connection_members
+            pass
+        else:
+            outList.append(valuesList)
+            columnsList.append(column)
+    return outList, columnsList
+
+
+def _recursive_getattr(topology, attr, attr_attr):
+    """Parse a topology to get a list of attributes from an iterable."""
+
+    def _getattr(obj, attr1):
+        try:
+            return getattr(obj, attr1)
+        except AttributeError as e:
+            raise AttributeError(
+                f"The GMSO Topology is missing the requested attribute {attr1} from {obj}.{attr_attr}"
+            )
+
+    iteritems = getattr(topology, attr)
+    parseFunction = lambda x: functools.reduce(
+        _getattr, [x] + attr_attr.split(".")
+    )
+    return list(map(parseFunction, iteritems))
+
+
+def _pandas_from_parameters(
+    topology, df, parameter, site_attrs=None, unyts_bool=True
+):
+    """Members of a connection attributes added to the dataframe.
+
+    Add to a pandas dataframe the site indices for each connection member in a
+    multimember topology attribute such as a bond. Also include information about
+    those sites in the site_attrs list.
+
+    """
+    if site_attrs is None:
+        site_attrs = []
+    sites_per_connection = len(getattr(self, parameter)[0].connection_members)
+    for site_index in np.arange(sites_per_connection):
+        df["Atom" + str(site_index)] = list(
+            str(connection.connection_members[site_index].name)
+            + f"({self.get_index(connection.connection_members[site_index])})"
+            for connection in getattr(self, parameter)
+        )
+    for attr in site_attrs:
+        df = self._parse_dataframe_attrs(
+            df, attr, parameter, sites_per_connection, unyts_bool
+        )
+    return df
+
+
+def _parse_dataframe_attrs(
```

Comment:
We could remove this one as well

Reply:
Yeah I left in a lot of the old bits as reference, but it's time to remove them.